src/Service/ServiceEndpoint.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Basis\Nats\Service;
+
+use Basis\Nats\Client;
+use Basis\Nats\Message\Payload;
+use Basis\Nats\Queue;
+
+class ServiceEndpoint
+{
+    private int $num_requests = 1;
+
+    private int $num_errors = 0;
+
+    private string $last_error = '';
+
+    private float $processing_time = 0;
+
+    private Client|Queue $subscription;
+
+    public function __construct(
+        private readonly Service $service,
+        private readonly string  $name,
+        private readonly string  $subject,
+        private                  $endpointHandler,
+        private readonly string  $queue_group = 'q'
```

Comment:
this library use another code style, see phpcs.xml and validate it using phpcs binary.
indentation using spaces should be removed